presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java

```diff
@@ -88,19 +94,24 @@ public synchronized Response install(@NotNull @ApiParam("Url to fetch package")
     public synchronized Response upgrade(@NotNull @ApiParam("Url to fetch package") String locationToFetchPackage,
             @QueryParam("checkDependencies") @DefaultValue("true") @ApiParam("If false, disables dependency checking") boolean checkDependencies)
     {
+        LOGGER.debug("POST /package  url: {} checkDependencies: {}", locationToFetchPackage, checkDependencies);
         try {
-            if (installedPackageType != PackageType.valueOf(getFileExtension(locationToFetchPackage).toUpperCase())) {
+            PackageType packageType = PackageType.valueOf(getFileExtension(locationToFetchPackage).toUpperCase());
+            if (installedPackageType != packageType) {
+                LOGGER.error("Expected package type {} , found {}", installedPackageType, packageType);
```

Comment:
The message should be like: `Unmatched package type` or something like `Installed using blablabla, but upgrade using blablabla`